src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  for (const item of items) {
+    const keyValue = item[key];
+    const keyValueStr = String(keyValue);
+
+    if (!result[keyValueStr]) {
+      result[keyValueStr] = [];
+    }
```

Comment:
The check for the existence of `result[keyValueStr]` is good. However, consider using `result.hasOwnProperty(keyValueStr)` for clarity and to avoid issues if the `keyValueStr` happens to be a property inherited from the prototype chain.